diff --git a/services_views.module b/services_views.module
index 70a86d1..667c96a 100755
--- a/services_views.module
+++ b/services_views.module
@@ -78,6 +78,14 @@ function services_views_services_resources() {
         'optional' => TRUE,
         'default value' => FALSE,
       ),
+      'filters' => array(
+        'name' => 'filters',
+        'type' => 'array',
+        'description' => 'A list of filters to pass to the view.  These are defined by the exposed filters on your view.  Example call: <code>/views/your_view?filters[nid]=12345</code>',
+        'source' => array('param' => 'filters'),
+        'optional' => TRUE,
+        'default value' => array(),
+      ),
     ),
   );
 
diff --git a/services_views.resource.inc b/services_views.resource.inc
index 4c26f62..e9905d2 100755
--- a/services_views.resource.inc
+++ b/services_views.resource.inc
@@ -19,17 +19,21 @@
  *   Array (optional). A list of arguments to pass to the view.
  * @param $return_type
  *   String (optional). Whether to return the raw data results (FALSE), the entire views object ('view') or themed results ('theme').
+ * @param $args
+ *   Array (optional). A list of exposed filters to pass to the view.
  *
  * @return
  *  Array. The views return.
  */
-function services_views_retrieve($view_name, $display_id = 'default', $args = array(), $offset = 0, $limit = 10, $return_type = FALSE) {
+function services_views_retrieve($view_name, $display_id = 'default', $args = array(), $offset = 0, $limit = 10, $return_type = FALSE, $filters = array()) {
   $result = array();
   $view = views_get_view($view_name);
 
-  // Put all arguments and then execute.
+  // Put all arguments and filters and then execute.
   $view->set_arguments($args, FALSE);
+  $view->set_exposed_input($filters);
   $view->set_offset($offset);
+
   // Support for Views 2
   if (method_exists($view, 'set_use_pager')) {
     // If offset is set we can't have a user pager.
